### ISSUE TITLE
Error in `node': free(): invalid pointer: 0x000001000a79f858 

### ISSUE THREAD

**Issue by seth-priya** [RuntimeTools/appmetrics#190]
Hi,

I am not sure if this is in anyway related to the issue reported earlier:
https://github.com/RuntimeTools/appmetrics/issues/186

I am observing an issue with the latest master branch on Ubuntu 15.10 (both x86 and ppc64le). I have not run into this on Ubuntu 15.04 yet or on Red Hat.

Basically, the tests are aborted and exit as soon as they start

Here is what I see on the console:
test@pts00449-vm29:~/appmetrics$ npm test

> appmetrics@1.0.8-dev.0 test /home/test/appmetrics
> node tests/test_runner.js

*** Error in `node': free(): invalid pointer: 0x000001000a79f858 ***
Aborted
npm ERR! Test failed.  See above for more details.

I have tried this with node version v4.1.1 and 0.12 and the following steps:
git clone https://github.com/RuntimeTools/appmetrics
cd appmetrics && npm install node-gyp nan@2.x && node_modules/.bin/node-gyp configure rebuild && cd .. && npm install ./appmetrics
cd appmetrics && npm test

Any inputs here would be of great help. Thanks!

Priya

**Comment by hhellyer** [comment]
I have seen an issue running on Ubuntu 15.10, I believe it's a linking problem with the appmetrics libraries.

**Comment by sam-github** [comment]
I have the same problem, @tobespc, on master, running `npm test`:

```
> appmetrics@1.0.9-dev.0 test /home/sam/w/ibm/appmetrics
> node tests/test_runner.js

*** Error in `node': free(): invalid pointer: 0x0000000003859658 ***
Aborted (core dumped)
npm ERR! Test failed.  See above for more details.
ibm/appmetrics (fix-install-assumptions *%) % uname -a
Linux samtu 4.2.0-35-generic #40-Ubuntu SMP Tue Mar 15 22:15:45 UTC 2016 x86_64 x86_64 x86_64 GNU/Linux
ibm/appmetrics (fix-install-assumptions *%) % node --version
v4.2.1
ibm/appmetrics (fix-install-assumptions *%) % npm --version
2.14.7
ibm/appmetrics (fix-install-assumptions *%) % g++ --version
g++ (Ubuntu 5.2.1-22ubuntu2) 5.2.1 20151010

ibm/appmetrics (fix-install-assumptions *%) % cat /etc/issue
Ubuntu 15.10 \n \l

```

**Comment by tobespc** [comment]
ok, we are testing a change for this at the moment, will update later

**Comment by tobespc** [comment]
Have a fix for this that will be released in the 1.0.9 version of appmetrics due out this week

**Comment by tobespc** [comment]
change made to agent core 3.0.9 level
